### PR TITLE
Reduce memory overhead in Filter by fixing alignment

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce memory overhead in Filter by fixing alignment
Risk Level: Low

Sizeof test:
Before changes: 536
After changes: 512

### CODE REVIEW

File: source/common/router/router.h [hunk 1]
```diff
@@ -648,20 +643,26 @@ class Filter : Logger::Loggable<Logger::Id::router>,
...
+  Network::Socket::OptionsSharedPtr upstream_options_;
+  // Set of ongoing shadow streams which have not yet received end stream.
+  absl::flat_hash_set<Http::AsyncClient::OngoingRequest*> shadow_streams_;
+
+  // Keep small members (bools and enums) at the end of class, to reduce alignment overhead.
+  uint32_t retry_shadow_buffer_limit_{std::numeric_limits<uint32_t>::max()};
+  uint32_t attempt_count_{1};
+  uint32_t pending_retries_{0};
+  Http::Code timeout_response_code_ = Http::Code::GatewayTimeout;
+  FilterUtility::HedgingParams hedging_params_;
```
Comment (adisuissa) [inline source/common/router/router.h:656]: Might also be good to add `: 1` to [hedge_on_per_try_timeout_](https://github.com/envoyproxy/envoy/blob/f5677953923766d9b101c798de3896d3504a60f4/source/common/router/router.h#L65)

**Review by adisuissa — COMMENTED** [review]
Thanks, left a couple of nits, but otherwise LGTM.

File: source/common/router/router.h (hunk 1 above)
Comment (DiazAlan) [reply source/common/router/router.h:657]: Done

File: source/common/router/router.h [hunk 2]
```diff
@@ -648,20 +643,26 @@ class Filter : Logger::Loggable<Logger::Id::router>,
...
+  bool grpc_request_{};
+  bool exclude_http_code_stats_ = false;
```
[the remark this answers is not in the document]
Comment (DiazAlan) [reply source/common/router/router.h]: Done

File: source/common/router/router.h [hunk 3]
```diff
@@ -364,10 +364,11 @@ class Filter : Logger::Loggable<Logger::Id::router>,
                public RouterFilterInterface {
 public:
   Filter(FilterConfig& config, FilterStats& stats)
-      : config_(config), stats_(stats), downstream_1xx_headers_encoded_(false),
-        downstream_response_started_(false), downstream_end_stream_(false), is_retry_(false),
-        request_buffer_overflowed_(false), streaming_shadows_(Runtime::runtimeFeatureEnabled(
-                                               "envoy.reloadable_features.streaming_shadow")) {}
+      : config_(config), stats_(stats), grpc_request_{}, exclude_http_code_stats_(false),
```
Comment (adisuissa) [inline source/common/router/router.h:367]: nit: to make it easier to read and consistent with other fields in this MIL
```suggestion
      : config_(config), stats_(stats), grpc_request_(false), exclude_http_code_stats_(false),
```

**Review by adisuissa — APPROVED** [review]
Thanks, left one small nit, but otherwise LGTM.

/assign-from @envoyproxy/senior-maintainers

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**